lib/teamsnap.rb

```diff
@@ -131,15 +131,24 @@ def init(opts = {})
       apply_endpoints(self, collection) && true
     end
 
-    def run(via, href, args = {})
+    def run(via, href, args = {}, opts = {})
       resp = client.send(via, href, args)
 
       if resp.status == 200
-        Oj.load(resp.body)
-          .fetch(:collection)
+        collection = Oj.load(resp.body).fetch(:collection)
+        if opts[:backup_cache]
```

Comment:
You could always spawn a thread to do the work for each call. Since it's all IO. Just call join once init is complete.